pyndustric/compiler.py

```diff
@@ -818,6 +830,16 @@ def emit_tuple_syscall(self, node: ast.Call, outputs: list):
                 self.ins_append(f"ulocate damaged core {enemy} @copper {output}")
             else:
                 raise CompilerError(ERR_BAD_SYSCALL_ARGS, node)
+
+        elif node.func.value.id == "Unit" and node.func.attr == "get_block":
+            if len(node.args) != 2:
+                raise CompilerError(ERR_BAD_SYSCALL_ARGS, node)
+            if len(outputs) != 2:
+                raise CompilerError(ERR_BAD_TUPLE_ASSIGN, node)
+            output = " ".join(outputs)
```

Comment:
Would be cool if this worked a bit like `Unit.locate`, which can return more or less information depending on the left side.

But I'm not sure if it should work like this:

```python
building_type, = Unit.get_block(x, y)
building_type, building = Unit.get_block(x, y)
```

Or like this:

```python
building, = Unit.get_block(x, y)
building_type, building = Unit.get_block(x, y)
```

The second would be a bit more confusing but perhaps it's more useful? What would the user mainly care about, the building type or the building? Perhaps we can flip the order (building, building type).

In any case I think we should at least offer the first option (optional building value).

Reply:
Done and done! I did the second option: it doesn't seem confusing and getting the building over its type makes more sense (since you can sensor a lot of its properties including type)